### PR TITLE
Add not_user_type param to the list accounts admin API

### PR DESCRIPTION
**Pull request by weeman1337**
closes https://github.com/matrix-org/synapse/issues/15720

### Pull Request Checklist

* [x] Pull request is based on the develop branch
* [x] Pull request includes a [changelog file](https://matrix-org.github.io/synapse/latest/development/contributing_guide.html#changelog). The entry should:
  - Be a short description of your change which makes sense to users. "Fixed a bug that prevented receiving messages from other servers." instead of "Moved X method from `EventStore` to `EventWorkerStore`.".
  - Use markdown where necessary, mostly for `code blocks`.
  - End with either a period (.) or an exclamation mark (!).
  - Start with a capital letter.
  - Feel free to credit yourself, by adding a sentence "Contributed by @github_username." or "Contributed by [Your Name]." to the end of the entry.
* [x] Pull request includes a [sign off](https://matrix-org.github.io/synapse/latest/development/contributing_guide.html#sign-off)
* [x] [Code style](https://matrix-org.github.io/synapse/latest/code_style.html) is correct
  (run the [linters](https://matrix-org.github.io/synapse/latest/development/contributing_guide.html#run-the-linters))

### CODE REVIEW

**Comment by weeman1337** [comment]
Open question: Should an index be added to `user_type`?

File: tests/rest/admin/test_user.py [hunk 1]
```diff
@@ -933,6 +933,75 @@ def test_filter_out_approved(self) -> None:
...
+
+        # Request and exclude bot and support users
+        test_user_type(
+            [self.admin_user, regular_user_id],
+            not_user_types=[UserTypes.BOT, UserTypes.SUPPORT],
+        )
+
+        # Request and exclude empty user types →  only expected the bot and support user
+        test_user_type([support_user_id, bot_user_id], not_user_types=[""])
+
```
Comment (H-Shay) [inline tests/rest/admin/test_user.py:1004]: It looks like from the code the empty user type can be combined with the other user types in the `not_user_types` list, e.g. here it would look like `not_user_types=[UserTypes.BOT, ""]`, would it be possible to add a case here testing the combination of an empty type and non-empty type?

Comment (weeman1337) [reply tests/rest/admin/test_user.py:1004]: Done. I've also added a test to cover a custom user type (means neither bot nor service).

**Review by H-Shay — APPROVED** [review]
LGTM, I am going to punt on the index question and ask you to drop it in #synapse-dev or the backend channel next week.

**Comment by weeman1337** [comment]
[It looks like nobody has complained](https://matrix.to/#/!vcyiEtMVHIhWXcJAfl:sw1v.org/$OJgkHlwmIsOERiBSfyzVGfCamoIg9_TONq-vOpMU04s?via=matrix.org&via=element.io&via=beeper.com)

Go @H-Shay ?

**Comment by clokep** [comment]
For the index, could attempt a query on matrix.org and see how long it takes? We can always add it afterward though, so no reason not to merge IMO.

**Comment by H-Shay** [comment]
> It looks like nobody has complained

Thanks for following up on that!

**Comment by H-Shay** [comment]
Merged #15844 into develop.
